backends/p4tools/CMakeLists.txt

```diff
@@ -34,6 +34,14 @@ project(P4TOOLS VERSION ${P4C_SEM_VERSION_STRING})
 # inherit FindLibGc.
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
+# We need a fairly recent version of Z3.
+set(Z3_MIN_VERSION "4.8.14")
+find_package(Z3 ${Z3_MIN_VERSION} REQUIRED)
+
+if(NOT DEFINED Z3_VERSION_STRING OR ${Z3_VERSION_STRING} VERSION_LESS ${Z3_MIN_VERSION})
+  message(FATAL_ERROR "The minimum required Z3 version is ${Z3_MIN_VERSION}. Has ${Z3_VERSION_STRING}.")
+endif()
+
```

Comment:
Now reviewing my code it seems the move of this fragment was not needed. I can revert it, but it seems more logical to me to actually have the dependencies searched in the top-level p4tools cmake so I am leaving it here for now.